File: go/1207_Unique_Number_of_Occurrences/1207_Unique_Number_of_Occurrences.go
func uniqueOccurrences(arr []int) bool {
    m := make(map[int]int)
    counts := make(map[int]bool)
    for _, v := range arr {
        m[v] += 1
    }
    for _, v := range m {
        _, exists := counts[v]
        if exists {
            return false
        } else {
            counts[v] = true
        }
    }
    return true
}

File: java/1207_Unique_Number_of_Occurrences/1207_Unique_Number_of_Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int v : arr) {
            map.merge(v, 1, Integer::sum);
        }
        HashSet<Integer> counts = new HashSet<Integer>();
        for (int key : map.values()) {
            if (counts.contains(key)) {
                return false;
            } else {
                counts.add(key);
            }
        }
        return true;
    }
}
